anesthetic/read/getdistreader.py

```diff
@@ -36,41 +36,21 @@ def paramnames(self):
         except IOError:
             return super().paramnames()
 
-    def limits(self):
-        """Read <root>.ranges in getdist format."""
-        try:
-            with open(self.ranges_file, 'r') as f:
-                limits = {}
-                for line in f:
-                    line = line.strip().split()
-                    paramname = line[0]
-                    try:
-                        xmin = float(line[1])
-                    except ValueError:
-                        xmin = None
-                    try:
-                        xmax = float(line[2])
-                    except ValueError:
-                        xmax = None
-                    limits[paramname] = (xmin, xmax)
-                return limits
-        except IOError:
-            return super().limits()
-
     def samples(self, burn_in=False):
         """Read <root>_1.txt in getdist format."""
         data = np.array([])
         for chains_file in self.chains_files:
             data_i = np.loadtxt(chains_file)
             if burn_in:
-                if 0 < burn_in < 1:
+                if isinstance(burn_in, float) and 0 < burn_in < 1:
                     index = int(len(data_i) * burn_in)
-                elif type(burn_in) is int and 1 < burn_in < len(data_i):
+                elif isinstance(burn_in, int) and 1 < burn_in < len(data_i):
```

Comment:
see above. Annoying this code is repeated -- hopefully something we'll resolve with a better inheritance hierarchy in the resolution to #208